modules/objdetect/src/qrcode.cpp

```diff
@@ -2251,6 +2251,80 @@ bool QRDecode::preparingCurvedQRCodes()
     return true;
 }
 
+/**
+ * @param finderPattern - 4 points of finder pattern markers, calculated by findPatternsVerticesPoints()
+ * @param cornerPointsQR - 4 corner points of QR code
+ * @return pair<int, int> first - the index in points of finderPattern closest to the corner of the QR code,
+ *                         second - the index in points of cornerPointsQR closest to the corner of finderPattern
+ *
+ * This function matches finder patterns to the corners of the QR code. Points of finder pattern calculated by
+ * findPatternsVerticesPoints() may be erroneous, so they are checked.
+ */
+static inline std::pair<int, int> matchPatternPoints(const vector<Point> &finderPattern,
+                                                    const vector<Point2f> cornerPointsQR){
+    // check that there are 4 points in the finderPattern
+    if (cornerPointsQR.size() != 4ull)
+        return std::make_pair(-1, -1);
+
+    float distanceToOrig = normL2Sqr<float>(Point2f(finderPattern[0]) - cornerPointsQR[0]);
+    int closestFinderPatternV = 0;
+    int closetOriginalV = 0;
+
+    // check that the sides are the same size
+    const float patternMaxRelativeLen = .3f;
+    vector<float> sidesLen;
+    for (size_t i = 0; i < finderPattern.size(); i++) {
+        sidesLen.push_back(sqrt(normL2Sqr<float>(Point2f(finderPattern[i] - finderPattern[(i+1ull)%4ull]))));
+        if (i > 0ull && (abs(sidesLen[i]/sidesLen[i-1]) - 1.f) > patternMaxRelativeLen)
```

Comment:
I suspect an issue in the condition.  `abs` applied to `sidesLen[i]/sidesLen[i-1]`. It's always >= 0.

Reply:
I propose to extract the loop as dedicated function and name it like `checkFinderPatternByAspect`. It makes code more readable. Also the function result depends on points order and it's strange. I propose to check delta between minimum and maximum length or deviation from average length.